engine.py

```diff
@@ -93,12 +107,41 @@ def get_bishop_moves(self, r, c, moves):
 
 	def get_knight_moves(self, r, c, moves):
  		##TODO
- 		pass
+	    pass
 
 
 	def get_king_moves(self, r, c, moves):
+		"""
+			calculates all possible king moves for a given colour (light or dark)
+			and appends them to a list
+
+			input parameters:
+			r     --> starting row (int)
+			c     --> starting column (int)
+			moves --> posiible moves container (list)
+
+			return parameter(s):
+			None
+ 		"""
  		##TODO
- 		pass
+		rows = (r-1, r, r+1)
+		columns = (c-1, c, c+1)
+		if self.light_to_move: # if it's light's turn to move
+			available_squares = (" ", "d") #if square is empty or has a dark piece
+			for i in range(len(self.board)):
+				for j in range(len(self.board[i])):
```

Comment:
Awesome work! code runs correctly. However it might need some refactoring for efficiency; right now the nested for loop checks every position on the board from top to bottom for a valid king move regardless of where king is. Rather, try checking for valid moves from the king's position outward.